### PR TITLE
Roadmap: topics to tackel for version 1

### PR DESCRIPTION
**Pull request by tsunammis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 **Tartiflette is under heavy development. [Feel free to join us to build Tartiflette](./docs/CONTRIBUTING.md).**
 
+That engine **IS NOT** ready to use, feel free to take a look of **[our road to v1](#roadmap---road-to-v1)**.
```
Comment (achedeuzot) [inline README.md]: The engine

take a look at our roadmap for v1

File: README.md [hunk 2]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
 
 You need to build and install this libgraphqlparser binary on your system before using this.
 See [here](https://github.com/graphql/libgraphqlparser) for more details.
+
+## Roadmap - Road to v1
+
+Based on our experience to managing GraphQL API in production for more than one year. We decided to work on a brand new GraphQL engine.
```
Comment (achedeuzot) [inline README.md]: Following on our experience **in** managing GraphQL API**s** in production **for more than a year**, we've decided to build a brand new GraphQL engine.

File: README.md [hunk 3]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+
+Here are the subjects we are working on:
+
+- [x] Define Open Source guidelines (Contributing, Code of conduct, Issue Template, PR Template).
+- [x] Build communication interface between [libgraphqlparser](https://github.com/graphql/libgraphqlparser) & Tartiflette through [CFFI](https://cffi.readthedocs.io).
+- [x] Build a `Parser` which parse a GraphQL Query and creates a list of Asynchronous Tasks.
```
Comment (achedeuzot) [inline README.md]: which parses

File: README.md [hunk 4]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+- [x] Build an `Executor` which execute the Async Tasks list created by the `GraphQL Query Parser`.
```
Comment (achedeuzot) [inline README.md]: which executes.

Use the same names: here you use `GraphQL Query Parser`, above, just `Parser`. Same comment for `Asynchronous Tasks` vs `Async Tasks`

File: README.md [hunk 5]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+- [x] Build a `Parser` which parse a [Schema Definition Language](https://github.com/facebook/graphql/blob/master/spec/Section%202%20--%20Language.md) and creates the associated types in Python object.
```
Comment (achedeuzot) [inline README.md]: parse**s** **the** SDL (the spec is known, so you can refer to it using "the", "a" would mean there are several.)

Comment (achedeuzot) [reply README.md]: and created the associated schema and types as Python objects

File: README.md [hunk 6]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+- [x] Bind the Types specified in the SDL to the `Executor`.
+- [ ] (Executor) Use `Field` instead `Node`.
```
Comment (achedeuzot) [inline README.md]: instead of

File: README.md [hunk 7]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+- [ ] (Doc) Describe the API that will be used by the Tartiflette users.
+  - [ ] (Doc) API - Engine constructor.
+  - [ ] (Doc) API - How to declare custom directive in the Engine Constructor.
```
Comment (achedeuzot) [inline README.md]: a custom directive OR custom directives

File: README.md [hunk 8]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+  - [ ] (Doc) API - How to declare middleware on resolver.
```
Comment (achedeuzot) [inline README.md]: on a resolver OR on resolvers

File: README.md [hunk 9]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+  - [ ] (Doc) API - Executor.
+  - [ ] (Doc) API - Resolver.
+  - [ ] (Doc) API - Resolver - `Info` parameter.
+- [ ] (Introspection) Implement the `__type` Field.
+- [ ] (Introspection) Implement the `__schema`Field.
+- [ ] (Resolver) Algorithm to apply [default resolver](https://github.com/dailymotion/tartiflette/blob/master/tartiflette/parser/visitor.py#L18)
```
Comment (achedeuzot) [inline README.md]: Unclear

File: README.md [hunk 10]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+- [ ] (Executor) Typing [resolver outputs](https://github.com/dailymotion/tartiflette/blob/master/tartiflette/parser/visitor.py#L191)
+- [ ] (Executor) Error management
+- [ ] (Executor) Abstract and Compound Types: Scalar, Unions, Interfaces, …
```
Comment (achedeuzot) [inline README.md]: Keep just Unions, Interfaces: Scalars are not abstract nor compound, my mistake in formulating this

File: README.md [hunk 11]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+- [ ] (Executor) Integrate the middleware in the Executor.
+- [ ] (CI) Run Code Quality checks + Tests
+- [ ] (CI) Automatize the integration of `libgraphqlparser`
+- [ ] (CI) Build & Publish artifact to pypi
+- [ ] (Website) Landing page for https://tartiflette.io
+- [ ] (Website) Expose documentation on https://tartiflette.io
+
+## Roadmap - Milestone #2
+
+- [ ] (SDL) Implement custom Scalar
```
Comment (achedeuzot) [inline README.md]: This is more or less already done :)

File: README.md [hunk 12]
```diff
@@ -14,3 +16,47 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+- [ ] Implement dynamic introspection based on directive
+- [ ] Think about `subscription`
```
Comment (achedeuzot) [inline README.md]: `subscription`s

File: README.md (hunk 1 above)
Comment (tsunammis) [reply README.md]: 👍

File: README.md (hunk 2 above)
Comment (tsunammis) [reply README.md]: 👍

File: README.md (hunk 3 above)
Comment (tsunammis) [reply README.md]: ok

File: README.md (hunk 4 above)
Comment (tsunammis) [reply README.md]: Ok

File: README.md (hunk 5 above)
Comment (tsunammis) [reply README.md]: ok

File: README.md [hunk 13]
```diff
@@ -14,3 +16,60 @@ This lib depends on libgraphqlparser (in submodule in this repository).
...
+* [ ] (API) Describe the API that will be used by the Tartiflette users.
+* [ ] (API) Engine constructor.
+* [ ] (API) How to declare custom directives in the Engine Constructor.
+* [ ] (API) How to declare middleware on a resolver.
+* [ ] (API) Executor.
+* [ ] (API) Resolver.
+* [ ] (API) Resolver - `Info` parameter.
+* [ ] (Website) Landing page for https://tartiflette.io
+* [ ] (Website) Expose documentation on https://tartiflette.io
+### Query Parser
```
Comment (abusi) [inline README.md]: Missing a space.

**Pull request merged**